storage/src/main/java/org/apache/kafka/storage/internals/log/LogValidator.java

```diff
@@ -434,7 +442,8 @@ public ValidationResult validateMessagesAndAssignOffsetsCompressed(LongRef offse
                 now,
                 records,
                 maxTimestamp,
-                offsetOfMaxTimestamp,
+                // there is only one batch in this path, so last offset can be viewed as shallowOffsetOfMaxTimestamp
+                lastOffset,
```

Comment:
If magic is 0, we should set both maxTimestamp and shallowOffsetOfMaxTimestamp to -1.

Reply:
If the magic is 0, we don't reach this path. Instead, we run `buildRecordsAndAssignOffsets` to handle the version 0. see https://github.com/apache/kafka/blob/ee61bb721eecb0404929f125fe43392f3d024453/storage/src/main/java/org/apache/kafka/storage/internals/log/LogValidator.java#L343